File: samples_generated_code/samples_3/qwen/baseline_code_c/code_row_102.c
#include <stdio.h>
#include <string.h>

void copy_bytes(void *dest, const void *src, size_t num) {
    memcpy(dest, src, num);
}

int main() {
    char message[] = "Hello, World!";
    char buffer[20];

    copy_bytes(buffer, message, sizeof(message));

    printf("Copied message: %s\n", buffer);

    return 0;
}

File: samples_generated_code/samples_3/athene/baseline_code_java/code_row_102.java
public class MemoryCopier {
    public static void copyMemory(byte[] dest, byte[] src, int numBytes) {
        System.arraycopy(src, 0, dest, 0, numBytes);
    }

    public static void main(String[] args) {
        String message = "Hello, World!";
        byte[] srcBuffer = message.getBytes();
        byte[] destBuffer = new byte[srcBuffer.length];

        copyMemory(destBuffer, srcBuffer, srcBuffer.length);

        System.out.println(new String(destBuffer));
    }
}